elastic/kibana
# [Security Solution][Detection Engine] deprecates signals migration APIs

## Summary

- addresses https://github.com/elastic/kibana/issues/195523
- deprecates https://www.elastic.co/guide/en/security/current/signals-migration-api.html APIs according to internal 9.x readiness [guideline](https://docs.google.com/document/d/1W7csjn6QYjrBjmbXMzSz_JUD4KcmWz8jTTtAWFwgUJM/edit?tab=t.0#heading=h.tui2zvb9gca6)
- adds deprecation feature in Kibana Upgrade Assistant. When non-migrated siem signals detected, it will be shown at a critical level
- adds snapshot telemetry
- while working on PR, discovered and fixed few issues in APIs
  - get migration status API did not work correctly with new `.alerts-*` indices, listing them as outdated
  - finalize migration API did account for spaces, when adding alias to migrated index
  - remove migration API failed due to lack of permissions to removed migration task from `.tasks` index

### How to test

#### How to create legacy siem index?

run script that used for FTR tests

```bash
node scripts/es_archiver --kibana-url=http://elastic:changeme@localhost:5601 --es-url=http://elastic:changeme@localhost:9200 load x-pack/test/functional/es_archives/signals/legacy_signals_index

node scripts/es_archiver --kibana-url=http://elastic:changeme@localhost:5601 --es-url=http://elastic:changeme@localhost:9200 load x-pack/test/functional/es_archives/signals/legacy_signals_index_another_1
```
These would create legacy siem indices. But be aware, it might break Kibana .alerts indices creation. But sufficient for testing

#### How to test deprecated APIs?

1. Run API https://www.elastic.co/guide/en/security/current/signals-migration-api.html
2. Observe warning deprecation on Kibana Upgrade page

<details>
<summary> Kibana Upgrade list of deprecations </summary>

<img width="2535" alt="Screenshot 2024-11-20 at 16 30 44" src="https://github.com/user-attachments/assets/7a79aac8-a414-4e9f-845f-8c354c8e58ef">

 </details>

#### How to test deprecated feature?
1. Observe critical  feature deprecation on Kibana Upgrade page, if you set up legacy siem signals

<details>
<summary> Kibana Upgrade feature deprecation flyout </summary>

<img width="800" alt="Screenshot 2024-11-20 at 16 32 35" src="https://github.com/user-attachments/assets/c255b766-047c-40c7-a170-a3fc5e4a90d3">

 </details>

#### How to test snapshot telemetry

Snapshot
For snapshot telemetry use [API](https://docs.elastic.dev/telemetry/collection/snapshot-telemetry#telemetry-usage-payload-api) call
OR
Check snapshots in Kibana adv settings -> Global Settings Tab -> Usage collection section -> Click on cluster data example link -> Check `legacy_siem_signals ` fields  in flyout

<details>
<summary> Snapshot telemetry </summary>

<img width="2516" alt="Screenshot 2024-11-20 at 16 38 16" src="https://github.com/user-attachments/assets/50d71eaf-a202-41bb-83ea-bd4a8e8b4d5e">


 </details>
